strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -376,79 +433,121 @@ def measure_dyne(self, covmat, indices, shots=1):
         Quantum Continuous Variables: A Primer of Theoretical Methods
         by Alessio Serafini page 129
         """
-        # if covmat.shape != (2 * len(indices), 2 * len(indices)):
-        #     raise ValueError("Covariance matrix size does not match indices provided")
-
-        # for i in indices:
-        #     if self.active[i] is None:
-        #         raise ValueError("Cannot apply homodyne measurement, mode does not exist")
-
-        # expind = np.concatenate((2 * np.array(indices), 2 * np.array(indices) + 1))
-        # mp = self.scovmat()
-        # (A, B, C) = ops.chop_in_blocks(mp, expind)
-        # V = A - np.dot(np.dot(B, np.linalg.inv(C + covmat)), np.transpose(B))
-        # V1 = ops.reassemble(V, expind)
-        # self.fromscovmat(V1)
-
-        # r = self.smean()
-        # (va, vc) = ops.chop_in_blocks_vector(r, expind)
-        # vm = np.random.multivariate_normal(vc, C, size=shots)
-        # # The next line is a hack in that it only updates conditioned on the first samples value
-        # # should still work if shots = 1
-        # va = va + np.dot(np.dot(B, np.linalg.inv(C + covmat)), vm[0] - vc)
-        # va = ops.reassemble_vector(va, expind)
-        # self.fromsmean(va)
-        return
+        if covmat.shape != (2 * len(indices), 2 * len(indices)):
+            raise ValueError("Covariance matrix size does not match indices provided")
+
```

Comment:
do we want to add a check to make sure covmat is a valid covariance matrix here ?

Reply:
Good idea. The user is currently not actually able to set covmat from the frontend.